sdk/devopsinfrastructure/azure-resourcemanager-devopsinfrastructure/src/main/java/com/azure/resourcemanager/devopsinfrastructure/fluent/PoolsClient.java

```diff
@@ -267,4 +269,30 @@ SyncPoller<PollResult<PoolInner>, PoolInner> beginUpdate(String resourceGroupNam
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
     PagedIterable<PoolInner> list(Context context);
+
+    /**
+     * Checks that the pool name is valid and is not already in use.
+     * 
+     * @param body The CheckAvailability request.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws com.azure.core.management.exception.ManagementException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the CheckNameAvailability operation response along with {@link Response}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    Response<CheckNameAvailabilityResultInner> checkNameAvailabilityWithResponse(CheckNameAvailability body,
```

Comment:
Do we know if the service is ready to release these new APIs added here?

Reply:
We won't release this update (or any change in mgmt libs in this PR or any other PR that not triggered by service request or service self-serve). The change would either release via their request, if the request is on this api-version, or been overridden if they request other api-version.